avalon/mongodb.py

```diff
@@ -390,17 +390,26 @@ def projects(self, query_filter=None, projection=None):
 
         find_args = [query_filter]
         if projection:
-            find_args.append(projection)
+            _projection = {"data.archived": 1}
+            _projection.update(projection)
+            find_args.append(_projection)
 
         for project_name in self._database.collection_names():
             if project_name in ("system.indexes",):
                 continue
 
             # Each collection will have exactly one project document
 
-            document = self._database[project_name].find_one(*find_args)
-            if document is not None:
-                yield document
+            doc = self._database[project_name].find_one(*find_args)
+            if doc is not None:
+
+                if no_archived and doc.get("data", {}).get("archived"):
```

Comment:
I though that key will be `active`?

Reply:
Yeah.., I actually struggled whether to use "active" or "archived" to align what it currently is for.

The reason I end up using "archive" instead of "active" is, the "active" seems to be an more general state, and "non-archived" is a sub-state that makes a project "active".

But I might be over thinking this, so please let know what you think :) And I am happy to change if "active" is good enough.

Reply:
I've opened the question in OpenPype PR...